docs/libcurl/opts/CURLOPT_SSLVERSION.3

```diff
@@ -50,6 +50,18 @@ TLSv1.1 (Added in 7.34.0)
 TLSv1.2 (Added in 7.34.0)
 .IP CURL_SSLVERSION_TLSv1_3
 TLSv1.3 (Added in 7.52.0)
+.IP CURL_SSLVERSION_MAX_DEFAULT
+Use as flag with CURL_SSLVERSION_TLSv1_x and behavior is same as
+CURL_SSLVERSION_MAX_TLSv1_2.
```

Comment:
I look to libraries and they are three kinds:
1. supports to get/set default maximum: nss
2. maximum is not determined: openssl, gnutls, mbedtls
3. doesn't support: darwinSSL, polarssl, gskit

For 1. default is determined by library.
For 2. what is default? eg: library supports TLSv3, but TLSv2 in this time is recommended.
For 3. default must be determined by us.

My suggestion:
For 1. Function <Curl_ssl_evaluate_version_max> (renamed retrieve_ssl_version_max) calls function <Curl_ssl_get_default_max_version> that returns CURL_SSLVERSION_MAX_TLSvX for certain ssl library.
For 2. I suggest use behavior as in point 3.
For 3. <Curl_ssl_get_default_max_version> returns CURL_SSLVERSION_MAX_TLSv1_2.

What do you think?